packages/react-components/react-input/src/components/Input/useInputStyles.styles.ts

```diff
@@ -28,7 +28,7 @@ const useRootClassName = makeResetStyles({
 
   // size: medium (default)
   minHeight: fieldHeights.medium,
-  padding: `0 ${tokens.spacingHorizontalMNudge}`,
+  padding: `0 ${tokens.spacingHorizontalXXS}`,
```

Comment:
This will still leave some dead space at the start and end of the input. It's not a big deal, but since you're going to the effort here, might as well move all of the padding to the input slot when there's no before/after content.
```suggestion
  padding: '0',
```

Then, you can add the combined padding to the `input` slot when it doesn't have before/after content. And when there is before/after content, then adjust the padding on the `root` and `input` slots appropriately.  See my main review comment for the padding values to use.